sk0-back/src/users/services/users/users.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@nestjs/common';
+import { CreateUserDto } from 'src/users/dtos/createUser.dto';
+import { User } from 'src/users/types/User';
+
+@Injectable()
+export class UsersService {
+  private users: User[] = [
+    {
+      id: 2,
+      username: 'testuser',
+      email: 'testuser@example.com',
+      password: 'password1234',
+    },
+  ];
+
+  createUser(userDto: CreateUserDto) {
+    this.users.push(userDto);
+  }
+
+  getUsers() {
+    return this.users;
+  }
+
+  getByEmail(email: string): User | undefined {
+    return this.users.find((user) => user.email === email);
+  }
```

Comment:
`findBy` (or `getBy`) should rather return a `null` instead of `undefined`.

https://codeburst.io/javascript-whats-the-difference-between-null-undefined-37793b5bfce6

Reply:
the service will be completely replaced after a git merge origin/dev - hope so